### ISSUE TITLE
Typescript examples

### ISSUE THREAD

**Issue by orouz** [react-spring/react-spring-examples#29]
I think it'd be great if the examples were written in typescript so it'd be clearer on how to use the two together. 

for example, i've adapted one of the examples to typescript (no changes, just `.tsx` instead of `.js`) but it seems to have typing errors. ([original](https://codesandbox.io/embed/1wqpz5mzqj), [fork](https://codesandbox.io/s/admiring-robinson-n7k36))

on my fork link you can see the file `index.tsx` is exactly as `index.js` but in it, the animated value of the `y` property is inferred as  `any`, which might indicate an issue with the lib's typings. i'm guessing if i had done something like `useTransition<A,B>` it'd work. but i'm not sure what `A` and `B` should be. from the types definition it seems like `A` is the provided item to iterate on and the `B` is the animated properties. but [that](https://codesandbox.io/s/sweet-aryabhata-kfz4s) didn't seem to work...

so basically, this boils down to:
1 - why does the example have wrongly inferred types? (`y` is `any`) 
2 - how could one utilize the generics parameters for (in this case) `useTransition` so it'd be properly typed? 

also, and more importantly, thank you for taking the time to even make these examples and awesome library! 
i'd be happy to help convert the examples to typescript once i figure out how to properly type stuff.

**Comment by orouz** [comment]
i'm closing this because it seems there's no interest in adding typescript examples. 
if it does become relevant, i'd definitely be willing to help, but as i said, i need some guidance with the typings.

**Comment by aleclarson** [comment]
The typings are completely rewritten in v9 beta. If you're still interested, you could try playing around with [the latest canary](https://spectrum.chat/react-spring/general/try-the-new-v9-canary~c5cad1eb-c6ff-4b40-9a10-76d368c3e404). If you have any questions, let me know in this thread. 👍 

> i'd be happy to help convert the examples to typescript once i figure out how to properly type stuff.

Sounds great, we would love the help :)

**Comment by lifeeric** [comment]
While learning React Sprint, I have to spend extra time to fix the type checking. we'd love it if there is an example with typescript.
